docs/hoon/demo.md

```diff
@@ -24,65 +25,60 @@ line is commented with its number:
     "Fizz"                                              ::  11
   ?:  =(0 (mod count 3))                                ::  12
     "Buzz"                                              ::  13
-  (scow %ud count)                                      ::  14
+  <count>                                               ::  14
 $(count (add 1 count))                                  ::  15
 ```
 
-Hoon is a functional language.  FizzBuzz is usually defined as a
-side effect, but officially Hoon has no side effects.  Accordingly,
-this code will *produce* a list of strings, not *print* a list of
-strings. (Of course, if you run this code from the Urbit
-[_Dojo_](../../using/shell) (shell) you will be able to see the
-product.)
+When FizzBuzz is written in non-functional languages the 
+output is generated as a side effect of the program, e.g., with 
+`printf()`.  Hoon is a functional language so technically it 
+has no side effects.  Accordingly, the code above *returns* a 
+list of strings, it doesn't print them. (If you run it from the 
+Urbit shell, [_Dojo_](../../using/shell), you will of course be 
+able to see the product.)
 
-The code produces a list in which each item is either a number
-(as a string), "Fizz", "Buzz", or "Fizzbuzz". The value `~` is
-"null" and indicates the end of the list.
+The code produces a list in which each item is either a number, 
+"Fizz", "Buzz", or "Fizzbuzz". The value `~` is "null" and indicates 
+the end of the list.
 
-Line 1 uses the _[rune](../../about/glossary#rune)_ `|=` (pronounced
-["bar-tis"](../syntax/#-glyphs-and-characters)) to define a function
-that takes one argument, labelled `end`. The argument must be of the
-type `@ud`, an unsigned decimal number.
+Line 1 uses the _[rune](../../about/glossary#rune)_ `|=` to define 
+a function that takes one input, `end`. The input value must be of 
+the type `@`, an atom (i.e., an unsigned integer).
 
-Line 2 uses the rune `=/` (pronounced "tis-fas") to declare a variable,
-`count` of the unsigned decimal number type, with initial value `1`.
+Line 2 uses `=/` to declare a variable, `count`, as an atom with an 
+initial value of `1`.
 
-Line 3 uses the rune `|-` ("bar-hep") to begin a loop.
+Line 3 uses `|-` to indicate the starting point of a loop 
+(technically a recursion).
```

Comment:
I'd lose the parenthetical. Technically, it's an immediately invoked `|.`, or one-armed core without a sample. The "recursion" is the re-evaluation of the arm of that core, which occurs on line 15.

Reply:
ok, dropped